### PR TITLE
Refactor pre add full sync options

### PR DESCRIPTION
**Pull request by marcserrat**
This is a work in progress PR, first phase is a refactor to be able to add new aspects of product syncronization

### CODE REVIEW

File: cnctcli/actions/products/sync.py [hunk 1]
```diff
@@ -3,168 +3,44 @@
...
-                except Exception as e:
-                    errors[row_idx] = [str(e)]
-        return (
-            skipped_count,
-            len(created_items),
-            len(updated_items),
-            len(deleted_items),
-            errors,
-        )
+        raise Exception("Not implemented")
```
Comment (ffaraoneim) [inline cnctcli/actions/products/sync.py:43]: Should raise NotImplementedError

File: cnctcli/actions/products/sync.py [hunk 2]
```diff
@@ -3,168 +3,44 @@
...
         self._wb = None
 
-    def open(self, input_file):
+    def open(self, input_file, worksheet):
         self._open_workbook(input_file)
-        if 'Items' not in self._wb.sheetnames:
-            raise ClickException('Invalid input file: File shall contain at least Items sheet.')
-        ws = self._wb[self._wb.sheetnames[0]]
+        if worksheet not in self._wb.sheetnames:
+            raise SkipSyncTask(f'File does not contain {worksheet} to synchronize, skipping')
```
Comment (ffaraoneim) [inline cnctcli/actions/products/sync.py:31]: Please change the name of this exception. The open should return the sheet called `worksheet` from the workbook to open. open should raise something like SheetNotFound. The caller of open should decide what to do if the sheet does not exist (continue or exit).

**Review by ffaraoneim — CHANGES_REQUESTED** [review]

File: cnctcli/actions/products/sync.py (hunk 1 above)
Comment (marcserrat) [reply cnctcli/actions/products/sync.py]: done

File: cnctcli/actions/products/sync.py (hunk 2 above)
Comment (marcserrat) [reply cnctcli/actions/products/sync.py]: done

**Pull request merged**